presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcPageSinkProvider.java

```diff
@@ -30,7 +30,7 @@
     private final JdbcClient jdbcClient;
 
     @Inject
-    public JdbcPageSinkProvider(@InternalBaseJdbc JdbcClient jdbcClient)
+    public JdbcPageSinkProvider(@StatsCollecting JdbcClient jdbcClient)
```

Comment:
... here it's really not obvious why we need stats-collecting instance.

we actually want something like `@TheClient` here.
Maybe _the client_ should be simply bound to `.bind(JdbcClient.class)`, without any annotation being part of the key

Reply:
same for `JdbcPageSinkProvider`, `JdbcSplitManager`

Reply:
> Maybe the client should be simply bound to .bind(JdbcClient.class), without any annotation being part of the key

This is how connector binds the `JdbcClient.class`, then in `base-jdbc` we need to bind it differently. Maybe there `@TheClient` (or even `@The` to use it for `ConnectionFactory`) is the way to go.

Reply:
or maybe `@Final`

Reply:
I can imagine `@Final` being used for different purposes, more closely aligned with `final` java keyword 

if we need an annotation `@TheClient` seems the best to me.
`@The` is cool, but may be confusing